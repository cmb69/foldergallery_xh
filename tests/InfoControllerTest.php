<?php

namespace Foldergallery;

use ApprovalTests\Approvals;
use Foldergallery\Infra\SystemChecker;
use Foldergallery\Infra\View;
use PHPUnit\Framework\TestCase;

class InfoControllerTest extends TestCase
{
    public function testRendersPluginInfo(): void
    {
        $systemChecker = $this->createMock(SystemChecker::class);
        $systemChecker->method("checkVersion")->willReturn(false);
        $systemChecker->method("checkExtension")->willReturn(false);
        $systemChecker->method("checkPlugin")->willReturn(false);
        $systemChecker->method("checkWritability")->willReturn(false);
        $plugin_tx = XH_includeVar("./languages/en.php", "plugin_tx");
        $view = new View("./views/", $plugin_tx["foldergallery"]);
        $sut = new InfoController("./plugins/foldergallery/", $systemChecker, $view);
        $response = $sut();
        $this->assertEquals("Foldergallery 1.0beta1", $response->title());
        Approvals::verifyHtml($response->output());
    }
}
