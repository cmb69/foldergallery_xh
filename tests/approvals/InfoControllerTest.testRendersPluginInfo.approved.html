<!-- foldergallery plugin info -->
<h1>Foldergallery 1.0beta1</h1>
<h2>System check</h2>
<div class="foldergallery_syscheck">
  <p class="xh_fail">the PHP version is at least 7.1.0: failure</p>
  <p class="xh_fail">PHP extension 'gd' is loaded: failure</p>
  <p class="xh_fail">PHP extension 'json' is loaded: failure</p>
  <p class="xh_warning">PHP extension 'exif' is loaded: warning</p>
  <p class="xh_fail">CMSimple_XH version is at least 1.7.0: failure</p>
  <p class="xh_fail">CMSimple_XH plugin 'jquery' is installed: failure</p>
  <p class="xh_warning">'./plugins/foldergallery/cache/' is writable: warning</p>
  <p class="xh_warning">'./plugins/foldergallery/config/' is writable: warning</p>
  <p class="xh_warning">'./plugins/foldergallery/css/' is writable: warning</p>
  <p class="xh_warning">'./plugins/foldergallery/languages/' is writable: warning</p>
</div>
